openpype/plugins/publish/validate_unique_subsets.py

```diff
@@ -49,17 +53,41 @@ def process(self, context):
                                  "{}".format(instance.name))
                 continue
 
-            instance_per_asset_subset[(asset, subset)].append(instance)
+            # for cases where multiple versions under same subset are published
+            version = instance.data.get("version")
+            if version is None:
+                # this might happen when workfile version sync is disabled
+                # we still need to validate cases where multiple "next version"
+                # instances are published under same subset
+                self.log.info("Instance found without `version` data: "
+                              "{}".format(instance.name))
```

Comment:
This will easily produce a cluttered artist facing log for which the artist knows nothing about what to do with this. It's not an actual issue - this is totally valid as far as I know. I would not log this actually. It's not a special case, also the check below in `else:` is also completely valid warbubg for the case where multiple instances with matching subset and asset include one that is `None` (actually, one might argue that if one has `None` and the others do not and their version numbers are higher than the current last version that the behavior is still dangerous and may be considered invalid/erroneous)